Userland/Libraries/LibWeb/HTML/HTMLFormControlsCollection.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2022, Johan Dahlin <jdahlin@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/HTML/HTMLFormControlsCollection.h>
+#include <LibWeb/HTML/Parser/HTMLParser.h>
+
+namespace Web::HTML {
+
+JS::NonnullGCPtr<HTMLFormControlsCollection> HTMLFormControlsCollection::create(DOM::ParentNode& root, Function<bool(DOM::Element const&)> filter)
+{
+    return *root.heap().allocate<HTMLFormControlsCollection>(root.realm(), root, move(filter));
+}
+
+HTMLFormControlsCollection::HTMLFormControlsCollection(DOM::ParentNode& root, Function<bool(DOM::Element const&)> filter)
+    : DOM::HTMLCollection(root, move(filter))
+{
+    set_prototype(&Bindings::cached_web_prototype(realm(), "HTMLFormControlsCollection"));
+}
+
+HTMLFormControlsCollection::~HTMLFormControlsCollection() = default;
+
+// https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#dom-htmlformcontrolscollection-nameditem
+Variant<JS::Handle<RadioNodeList>, JS::Handle<DOM::Element>, Empty> HTMLFormControlsCollection::named_item(FlyString const& name)
+{
+    // 1. If name is the empty string, return null and stop the algorithm.
+    if (name.is_empty())
+        return {};
+
+    // 2. If, at the time the method is called, there is exactly one node in the collection
+    // that has either an id attribute or a name attribute equal to name, then return that node and stop the algorithm.
```

Comment:
I still think the spec means something different here, but I guess this functions is still kind of TODO so perhaps it's fine

Reply:
I've read it over a couple of times, how would you interpret it differently? AFAIU, root()->children() will give the current live collection. The spec mentions `The HTMLFormControlsCollection interface is used for collections of listed elements in form elements.`, so is_listed() has to be true for an element to be considered part of the collection.

Reply:
I think that if there are multiple elements in the collection but just one has a name/id equal to `name` then it should return just that element. But I could also be misunderstand it, it is a bit ambiguous

Reply:
I see, that is how I interpret the spec as well. Multiple elements is covered in step 4, which is marked as a TODO as it also involved RadioNodeList to contain these results. I plan to spend some time writing a few good test cases and try to implement the remaining parts if I can in a follow up PR.